src/main/java/org/spongepowered/api/state/State.java

```diff
@@ -35,7 +36,7 @@
 import java.util.Optional;
 import java.util.function.Supplier;
 
-public interface State<S extends State<S>> extends SerializableDataHolder.Immutable<S> {
+public interface State<S extends State<S>> extends SerializableDataHolder.Immutable<S>, DataSerializable {
```

Comment:
Would it make sense to have `State<S extends State<S> & DataSerializable>` instead? I don't want the rabbit hole of the state being serializable by itself (unless we want